src/main/kotlin/uk/gov/justice/digital/hmpps/educationandworkplanapi/app/service/TelemetryService.kt

```diff
@@ -45,38 +45,6 @@ class TelemetryService(
     }
   }
 
-  /**
-   * Returns a map of data representing the custom dimensions for the `goal-create` event.
-   */
-  private fun createEventCustomDimensions(goal: Goal): Map<String, String> =
-    with(goal) {
-      mapOf(
-        "status" to status.name,
-        "stepCount" to steps.size.toString(),
-        "reference" to reference.toString(),
-        "notesCharacterCount" to (notes?.length ?: 0).toString(),
-      )
-    }
-
-  /**
-   * Returns a map of data representing the custom dimensions for the `goal-update` event.
-   */
-  private fun updateEventCustomDimensions(goal: Goal): Map<String, String> =
-    with(goal) {
-      mapOf(
-        "reference" to reference.toString(),
-        "notesCharacterCount" to (notes?.length ?: 0).toString(),
-      )
-    }
-
-  /**
-   * Returns a map of data representing the custom dimensions for the `step-remove` event.
-   */
-  private fun stepRemoveEventCustomDimensions(goal: Goal): Map<String, String> =
-    with(goal) {
-      mapOf(
-        "reference" to reference.toString(),
-        "stepCount" to steps.size.toString(),
-      )
-    }
+  private fun sendTelemetryEventForGoal(goal: Goal, telemetryEventType: TelemetryEventType) =
```

Comment:
This is the one and only place where we actually send the tracking event via the `telemetryClient`. Using the passed in `TelemetryEventType` it has everything it needs in order to send a telemetry tracking event to App Insights 👍